test/built-ins/Function/prototype/caller-arguments/strict-vs-nonstrict.js

```diff
@@ -0,0 +1,18 @@
+/*---
+description: Function properties behavior in strict vs non-strict contexts
+flags: [noStrict]
+---*/
+
+function nonStrictFunc() {
+    return nonStrictFunc.caller;
+}
+
+function strictFunc() {
+    return strictFunc.caller;
+}
+
+assert(!Object.hasOwnProperty.call(nonStrictFunc, "caller"), "non-strict function should not have own caller property");
+assert(!Object.hasOwnProperty.call(strictFunc, "caller"), "strict function should not have own caller property");
```

Comment:
The strict-mode case is already covered by:
- https://github.com/tc39/test262/blob/main/test/built-ins/Function/StrictFunction_restricted-properties.js